### DISCUSSION TITLE
Duplicate with relations?

### DISCUSSION THREAD

**Discussion by notflip** [prisma/prisma#19832] [Q&A] [answered]
I have the following database layout: a Survey has SurveyCategories (which have translations) and SurveyQuestions (which have translations). So I've got the following tables: `surveys, survey_categories, survey_category_translations, survey_questions, survey_question_translations`.

**Now at some point I need to duplicate this Survey, with all it's categories, questions and translations.**

What's a good way to go about this using Prisma? Thanks

**Comment by ludralph** [comment]
Hi @notflip 👋 

Can you please clarify what you mean by "duplicating this Survey, with all its categories, questions, and translations"? Do you want to duplicate the records in the database?

**Reply by notflip** [reply]
Yep!

**Reply by ludralph** [reply, marked as answer]
Hi @notflip 👋 

You would need to retrieve the survey you want to duplicate, including its categories, questions, and translations. And then, you would use the retrieved data to create a new survey. You also need to iterate over the categories, questions, and translations to create new records for them.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by notflip** [reply]
Is there not an easier way to do this? Besides looping the whole structure? Thanks @ludralph

**Reply by nurul3101** [reply]
I can't think of any other way besides cloning the existing records.
